internal/terraform/node_resource_abstract_instance.go

```diff
@@ -645,19 +647,69 @@ func (n *NodeAbstractResourceInstance) plan(
 	plannedChange *plans.ResourceInstanceChange,
 	currentState *states.ResourceInstanceObject,
 	createBeforeDestroy bool,
-	forceReplace []addrs.AbsResourceInstance) (*plans.ResourceInstanceChange, *states.ResourceInstanceObject, instances.RepetitionData, tfdiags.Diagnostics) {
+	forceReplace []addrs.AbsResourceInstance,
+	generateConfig bool,
+) (*plans.ResourceInstanceChange, *states.ResourceInstanceObject, instances.RepetitionData, tfdiags.Diagnostics) {
 	var diags tfdiags.Diagnostics
 	var state *states.ResourceInstanceObject
 	var plan *plans.ResourceInstanceChange
 	var keyData instances.RepetitionData
+	var generatedConfig *configs.Resource
 
-	config := *n.Config
 	resource := n.Addr.Resource.Resource
 	provider, providerSchema, err := getProvider(ctx, n.ResolvedProvider)
 	if err != nil {
 		return plan, state, keyData, diags.Append(err)
 	}
 
+	if providerSchema == nil {
+		diags = diags.Append(fmt.Errorf("provider schema is unavailable for %s", n.Addr))
+		return plan, state, keyData, diags
+	}
+	schema, _ := providerSchema.SchemaForResourceAddr(resource)
+	if schema == nil {
+		// Should be caught during validation, so we don't bother with a pretty error here
+		diags = diags.Append(fmt.Errorf("provider does not support resource type %q", resource.Type))
+		return plan, state, keyData, diags
+	}
+
+	// If we're importing and generating config, generate it now.
+	var generatedHCL string
+	if n.Config == nil {
```

Comment:
I don't think this can ever by `nil` for `import`, because this node can't exist during plan without there first being a configuration to create it. The original idea was that there would be a synthetic configuration created for import purposes when one didn't exist, even though that may need to be supplemented by generated attributes during the import process. While we may not consider supplementing existing config as in scope, we may also need to take into account the possibility of config generation when a real partial config exists too.

Reply:
Thanks! I missed that transformer piece, which I think does work. I can't recall if this can still be reached when there is no config for other reason however. It might still be safer to only use `generateConfig` to flag the config generation rather than the lack of config since this method is also used as part of the `apply` process.

Because there is no real config, this is also going to run into missing schemas and providers for implied providers. I think the provider transformers are the only other one which needs to the config, but we'll have to check more carefully.

Reply:
I've swapped the order of these checks round, so it checks for `generateConfig` and then panics if config is already set.